src/sdk/namespace_fs.js

```diff
@@ -1926,7 +1926,9 @@ class NamespaceFS {
             await this._check_path_in_bucket_boundaries(fs_context, file_path);
             dbg.log0('NamespaceFS: delete_object', file_path);
             let res;
-            if (this._is_versioning_disabled()) {
+            const is_key_dir_path = await this._is_key_dir_path(fs_context, params.key);
+            if (this._is_versioning_disabled() || is_key_dir_path) {
+                // TODO : Dir versioning not implemented, This should be fixed as part of #8320
```

Comment:
```suggestion
            // TODO- Directory object (key/) is currently can't co-exist while key (without slash) exists. see -https://github.com/noobaa/noobaa-core/issues/8320
            // Also, Directory object (key/) is currently not supported combined with versioning - see - <add link>
            if (this._is_versioning_disabled() || is_key_dir_path) {
```

please mention the 2 issues here - https://github.com/noobaa/noobaa-core/issues/8531